src/api/layers/WMSCapabilitiesParser.class.js

```diff
@@ -11,96 +11,100 @@ import proj4 from 'proj4'
 export default class WMSCapabilitiesParser {
     constructor(content, originUrl) {
         const parser = new WMSCapabilities()
-        Object.assign(this, parser.read(content))
+        try {
+            Object.assign(this, parser.read(content))
+        } catch (error) {
+            log.error(`Failed to parse capabilities of ${originUrl}`, error)
+            throw new Error(`Failed to parse WMTS Capabilities: invalid content`)
+        }
+
         this.originUrl = new URL(originUrl)
     }
 
     /**
      * Find recursively in the capabilities the matching layer ID node
      *
      * @param {string} layerId Layer ID to search for
-     * @param {WMSCapabilities.Capability.Layer} startFrom Layer node to start from
-     * @returns {WMSCapabilities.Capability.Layer} Capability layer node
+     * @returns {{
+     *     layer: WMSCapabilities.Capability.Layer
+     *     parents: [WMSCapabilities.Capability.Layer]
+     * }}
+     *   Capability layer node and its parents or an empty object if not found
      */
-    findLayer(layerId, startFrom = null) {
-        let layer = null
-        let layers = this.Capability.Layer.Layer
-        if (startFrom) {
-            layers = startFrom
-        }
-
-        for (let i = 0; i < layers.length && !layer; i++) {
-            if (layers[i].Name === layerId) {
-                layer = layers[i]
-            } else if (!layers[i].Name && layers[i].Title === layerId) {
-                layer = layers[i]
+    findLayer(layerId) {
+        return this._findLayer(layerId, this.Capability.Layer.Layer, [this.Capability.Layer])
+    }
+
+    _findLayer(layerId, startFrom, parents) {
+        let found = {}
+        const layers = startFrom
+
+        for (let i = 0; i < layers.length && !found.layer; i++) {
+            if (layers[i].Name === layerId || layers[i].Title === layerId) {
+                found.layer = layers[i]
+                found.parents = parents
             } else if (layers[i].Layer?.length > 0) {
-                layer = this.findLayer(layerId, layers[i].Layer)
+                found = this._findLayer(layerId, layers[i].Layer, [layers[i], ...parents])
             }
         }
-        return layer
+        return found
     }
```

Comment:
it really not a big deal/improvement, but I see you do not use any `this.xyz` in this function, so if you really wanted to hide it, you could declare it above the class definition as a function. As this function will not be exported it will be hidden to other files 😉

Reply:
Done